fiftyone/utils/openimages.py

```diff
@@ -0,0 +1,1325 @@
+    dataset.export(
+        export_dir, dataset_type=fot.FiftyOneDataset, export_media=False
+    )
+    num_samples = len(dataset)
+    dataset.delete()
+
+    return num_samples, all_classes
+
+
+def get_attributes(dataset_dir=None, scratch_dir=None):
+    """Gets the list of relationship attributes in the Open Images V6 dataset.
+
+    Args:
+        dataset_dir (None): the root directory the in which the dataset is
+            downloaded
+        scratch_dir (None): scratch directory used to download
+            attributes file if it is not available in ``metadata.json`` in the
+            ``dataset_dir`` splits
+
+    Returns:
+        a sorted list of attribute names
+    """
+    for split in _DEFAULT_SPLITS:
+        if dataset_dir:
+            metadata_path = os.path.join(dataset_dir, split, "metadata.json")
+        else:
+            metadata_path = os.path.join(
+                fo.config.dataset_zoo_dir,
+                "open-images-v6",
+                split,
+                "metadata.json",
+            )
+
+        if os.path.exists(metadata_path):
+            metadata = etas.load_json(metadata_path)
+            if (
+                "info" in metadata.keys()
+                and "attributes" in metadata["info"].keys()
+            ):
+                return metadata["info"]["attributes"]
+
+    if not scratch_dir:
+        if not dataset_dir:
+            scratch_dir = os.path.join(
+                fo.config.dataset_zoo_dir, "open-images-v6", "tmp-download"
+            )
+
+        else:
+            scratch_dir = os.path.join(dataset_dir, "tmp-download")
+
+    attrs_map = _get_attrs_map(
+        dataset_dir=dataset_dir, scratch_dir=scratch_dir
+    )
+    return sorted(list(attrs_map.values()))
+
+
+def get_classes(dataset_dir=None, scratch_dir=None):
+    """Gets the 601 boxable classes that exist in classifications, detections,
+    and relationships in the Open Images V6 dataset.
+
+    Args:
+        dataset_dir (None): the root directory the in which the dataset is
+            downloaded and ``info.json`` is stored
+        scratch_dir (None): scratch directory used to download classes file if
+            it is not available in ``info.json`` in the ``dataset_dir``
+
+    Returns:
+        a sorted list of class name strings
+    """
+    for split in _DEFAULT_SPLITS:
+        if dataset_dir:
+            metadata_path = os.path.join(dataset_dir, split, "metadata.json")
+        else:
+            metadata_path = os.path.join(
+                fo.config.dataset_zoo_dir,
+                "open-images-v6",
+                split,
+                "metadata.json",
+            )
+
+        if os.path.exists(metadata_path):
+            metadata = etas.load_json(metadata_path)
+            if (
+                "info" in metadata.keys()
+                and "classes" in metadata["info"].keys()
+            ):
+                return metadata["info"]["classes"]
```

Comment:
There's a fair amount of code duplication in these methods when reading from `metadata["info"]`. You could refactor into a method like `_load_metadata_if_possible()` and call it from each public method here